templates/definition/meter/kostal-plenticore.yaml

```diff
@@ -75,16 +75,46 @@ render: |
     source: sunspec
     {{- include "modbus" . | indent 2 }}
     value: 802:SoC # 802 battery control
-  limitsoc:
+  batterymode:
     source: watchdog
     timeout: {{ .watchdog }} # re-write at timeout/2
+    reset: 1 # reset watchdog on normal
     set:
-      source: modbus
-      {{- include "modbus" . | indent 4 }}
-      register:
-        address: 1042 # limit soc
-        type: writemultiple
-        encoding: {{ if (eq .endianness "big") }}float32{{ else }}float32s{{ end }}
+      source: switch
+      switch:
+      - case: 1 # normal
+        set: 
+          source: const
+          value: 0 # % to reset grid loading
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 1028 # battery charge current (DC) setpoint, relative
+              type: writemultiple
+              encoding: {{ if (eq .endianness "big") }}float32{{ else }}float32s{{ end }}
+      - case: 2 # hold
+        set:
+          source: const
+          value: 0 # % lock battery
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 1028 # battery charge current (DC) setpoint, relative
+              type: writemultiple
+              encoding: {{ if (eq .endianness "big") }}float32{{ else }}float32s{{ end }}
+      - case: 3 # charge
+        set:
+          source: const
+          value: -100 # % charge battery with maximum allowed power
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 1028 # battery charge current (DC) setpoint, relative
+              type: writemultiple
+              encoding: {{ if (eq .endianness "big") }}float32{{ else }}float32s{{ end }}
```

Comment:
Würde es hier nicht Sinn machen, noch `1044` mit `maxsoc` zu beschreiben?

Reply:
Ich gehe davon aus, das wäre hier mit der EVCC Steuerung redundant und bevorzugte das Schreibe weniger Register. Man müsste auch prüfen, ob das an anderen Stellen störend wirkt - auch wenn der maxSoC hier eindeutig sein sollte.
Interessant wird es, wenn man einen abweichenden maxSoC für die Netzladungsfunktion (Fall 3) in Betracht ziehen würde. Das bedingt aber meiner Einschätzung nach auch das zurücksetzen des Wertes für die anderen Fälle (1/2). Der Wert sollte dann ein eigener Konfigurationsparameter werden? Ich hab diesen Request mit der Intention gestellt, die Netzladung zu ermöglichen und dabei die Sperrlogik zugunsten weniger Register optimiert. 
Ein maxSoC für die Netzladung wäre ein anderer Schritt und eine neue Diskussion wert?

Reply:
Der `maxsoc` wird aber woanders von evcc nicht geschrieben. Insofern müsste er dann komplett aus dem Template raus oder?

Reply:
Ich denke nicht, denn der Konfigurationsparamter liegt außerhalb des Blocks für die Batteriesteuerung.
Das Thema mit dem maxSoC könnte in Diskussion #15747 weiter besprochen werden.

Reply:
minsoc/maxsoc braucht es nur wenn mit limitsoc gesteuert werden soll. Hier könnte das raus.